File: CleverFerretV2/feature/radio/src/main/java/com/cleverferret/v2/feature/radio/api/RadioFeatureApi.java
package com.cleverferret.v2.feature.radio.api;

import com.cleverferret.v2.core.common.result.IntegrationResult;
import com.cleverferret.v2.feature.radio.services.radio.RadioStationResultV1;

public interface RadioFeatureApi {
    String featureKey();

    IntegrationResult<RadioStationResultV1> resolveStation(String stationQuery);
}

File: CleverFerretV2/feature/radio/src/main/java/com/cleverferret/v2/feature/radio/api/RadioFeatureApi.kt
package com.cleverferret.v2.feature.radio.api

import com.cleverferret.v2.core.common.result.IntegrationResult
import com.cleverferret.v2.feature.radio.services.radio.RadioStationResultV1

interface RadioFeatureApi {
    fun featureKey(): String
    fun resolveStation(stationQuery: String): IntegrationResult<RadioStationResultV1>
}
